relational_operators/CreateIndexOperator.cpp

```diff
@@ -28,11 +28,15 @@ bool CreateIndexOperator::getAllWorkOrders(WorkOrdersContainer *container,
                                            StorageManager *storage_manager,
                                            const tmb::client_id foreman_client_id,
                                            tmb::MessageBus *bus) {
+  if (!work_generated_) {
+    work_generated_ = true;
+    relation_->addIndex(index_name_, index_descriptions_);
+  }
```

Comment:
Bug! Remove all code in the `if` statement, as it is duplicated in 
`CreateIndexOperator::updateCatalogOnCompletion` below.

Don't you have a test to cover that one relation is added the same index twice?

Finally, you may want to mark the `catalog` with a new `CatalogDatabase::Status` called `pendingIndexAdditions` (defined in `/catalog/CatalogDatabase.hpp`) using `CatalogDatabase::setStatus`.